File: 136_Single_Number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0
        for num in nums:
            ans ^= num
        return ans
        
# beats 76.69% runtime and 48.74% memory
# O(n) and O(1)
# https://leetcode.com/problems/single-number/submissions/1266508462/

File: 136_Single_Number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int ans = 0;
        for (auto num:nums)
            ans ^= num;
        return ans;
    }
};

// beats 88.01% runtime and 98.96% memory
// O(n) and O(1)
// https://leetcode.com/problems/single-number/submissions/1266511395/
